targets/TARGET_Realtek/TARGET_AMEBA/TARGET_RTL8195A/serial_api.c

```diff
@@ -271,30 +275,64 @@ static void SerialRxDoneCallBack(VOID *pAdapter)
     }
 }
 
+
+#ifdef CONFIG_MBED_ENABLED
+static void serial_loguart_irq_handler(uint32_t id, LOG_UART_INT_ID event)
+{
+    if (event == IIR_RX_RDY || event == IIR_CHAR_TIMEOUT)
+    {
+        if (log_irq_handler) {
+            log_irq_handler(serial_log_irq_ids, RxIrq);
+            }
+    } else if (event == IIR_THR_EMPTY) {
+        if (log_irq_handler) {
+                log_irq_handler(serial_log_irq_ids, TxIrq);
+            }
+    }
+    return;
+}
+#endif
+
+
+
 void serial_irq_handler(serial_t *obj, uart_irq_handler handler, uint32_t id) 
 {
+
+#ifdef CONFIG_MBED_ENABLED
```

Comment:
why `CONFIG_MBED_ENABLED` is here needed?

Reply:
I have just updated the description

Reply:
Thanks

> The "CONFIG_MBED_ENABLED" macro has been put so that the software is compatible with our own internal SDK.

I can see it is used in this file. I just do not follow how is it used. If config is enabled then what changes ? Why serial_api.c depends on the config enabled?

Reply:
We are reusing the same file in our internal SDK and within realtek, we are trying to maintain one single repository structure for all projects so when we build our SDK, it would refer to this same file. Our loguart is slightly different in nature and hence the changes